File: crossecore-csharp/Java/Java/Java_/BlockCommentBase.cs
/* CrossEcore is a cross-platform modeling framework that generates C#, TypeScript, 
 * JavaScript, Swift code from Ecore models with embedded OCL (http://www.crossecore.org/).
 * The original Eclipse Modeling Framework is available at https://www.eclipse.org/modeling/emf/.
 * 
 * contributor: Simon Schwichtenberg
 */
 
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using oclstdlib;
using Ecore;
namespace Java_{
	public class BlockCommentBase 
	:CommentImpl, BlockComment
	{
		
		protected override EClass eStaticClass() {
			return Java_PackageImpl.Literals.BLOCKCOMMENT;
		}
		
		
		
		public override Object eGet(int featureID, bool resolve, bool coreType) {
			switch (featureID) {
				case Java_PackageImpl.BLOCKCOMMENT_COMMENTS:
					return comments;
				case Java_PackageImpl.BLOCKCOMMENT_ORIGINALCOMPILATIONUNIT:
					return originalCompilationUnit;
				case Java_PackageImpl.BLOCKCOMMENT_ORIGINALCLASSFILE:
					return originalClassFile;
				case Java_PackageImpl.BLOCKCOMMENT_CONTENT:
					return content;
				case Java_PackageImpl.BLOCKCOMMENT_ENCLOSEDBYPARENT:
					return enclosedByParent;
				case Java_PackageImpl.BLOCKCOMMENT_PREFIXOFPARENT:
					return prefixOfParent;
			}
			return base.eGet(featureID, resolve, coreType);
		}
		
		
		public override void eSet(int featureID, object newValue) {
			switch (featureID) {
				case Java_PackageImpl.BLOCKCOMMENT_COMMENTS:
					comments.Clear();
					comments.AddRange(((List<EObject>)newValue)?.Cast<Comment>());
					return;
				case Java_PackageImpl.BLOCKCOMMENT_ORIGINALCOMPILATIONUNIT:
					originalCompilationUnit = (CompilationUnit) newValue;
					return;
				case Java_PackageImpl.BLOCKCOMMENT_ORIGINALCLASSFILE:
					originalClassFile = (ClassFile) newValue;
					return;
				case Java_PackageImpl.BLOCKCOMMENT_CONTENT:
					content = (string) newValue;
					return;
				case Java_PackageImpl.BLOCKCOMMENT_ENCLOSEDBYPARENT:
					enclosedByParent = (bool) newValue;
					return;
				case Java_PackageImpl.BLOCKCOMMENT_PREFIXOFPARENT:
					prefixOfParent = (bool) newValue;
					return;
			}
			base.eSet(featureID, newValue);
		}
		
		
	}
}

File: crossecore-typescript/java_/BlockCommentBase.ts
import {EClass} from "ecore/EClass";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {BlockComment} from "./BlockComment";
import {CommentImpl} from "./CommentImpl";
		
			export class BlockCommentBase
			extends CommentImpl
			implements BlockComment
			{

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return BlockCommentBase.eStaticClass;
				}
			
			
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.BLOCKCOMMENT_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.BLOCKCOMMENT_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.BLOCKCOMMENT_ORIGINALCLASSFILE:
							return this.originalClassFile;
						case Java_PackageLiterals.BLOCKCOMMENT_CONTENT:
							return this.content;
						case Java_PackageLiterals.BLOCKCOMMENT_ENCLOSEDBYPARENT:
							return this.enclosedByParent;
						case Java_PackageLiterals.BLOCKCOMMENT_PREFIXOFPARENT:
							return this.prefixOfParent;
					}
					//return this.eGetFromComment(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
